### PR TITLE
Add redundant Class Name test

### PR DESCRIPTION
**Pull request by TylorS**
After seeing the issue for Cycle-DOM that multiples of the same className was being added, I figured it was highly likely that the same bug would exist here. After adding the test it was readily apparent that the issue did in fact occur.

After adding the `if` statement to ensure that would not happen, I was running into an assertion error involving the className and at first I thought it was some kind of snabbdom patch error so I cleaned up the `rootElem$` to make sure that it was the patched DOM node getting checked.

When that did not fix it, I realized that `h(sel, data, children)` expects `sel` to be a string without spaces to add id's and className's, we were using `vtree.sel` impoperly by using a space.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -31,8 +31,10 @@ const isolateSink =
   (sink, scope) =>
     sink.map(
       vtree => {
-        const c = `${vtree.sel} cycle-scope-${scope}`.trim()
-        vtree.sel = c
+        if (vtree.sel.indexOf(`cycle-scope-${scope}`) === -1) {
+          const c = `${vtree.sel}.cycle-scope-${scope}`.trim()
```
Comment (staltz) [inline src/index.js]: You might not need to trim() here since `sel` is not a className, and classNames are the ones that have spaces around.

Comment (staltz) [inline src/index.js]: Removing trim() might bump up perf a bit.

Comment (TylorS) [inline src/index.js]: Good catch!

**Comment by TylorS** [comment]
Removed and tests still passing. Thanks @staltz

**Comment by staltz** [comment]
:+1:

File: src/index.js [hunk 2]
```diff
@@ -150,20 +152,23 @@ const makeDOMDriver =
...
+                  const vnode = patch(prevView, newView)
+                  add(vnode.elm)
+                  return vnode
+                },
+                rootElem
+              )
+              .then(end)
+              .catch(error)
+          }
+        )
```
Comment (Frikki) [inline src/index.js]: Why these changes? Did the test require these changes?

Comment (TylorS) [inline src/index.js]: Technically no, but it got changed along the way as I was trying to figure out where some problems were coming from. At first I thought `add(newView.elm)` might be pushing the wrong Element. I also cleaned up based on things I've learned along the way in that process. 

I can revert these changes and submit it as a separate PR
